build/README.md

```diff
@@ -0,0 +1,164 @@
+# Building fms-hf-tuning as an Image
+
+The Dockerfile provides a way of running fms-hf-tuning SFT Trainer. It installs the dependencies needed and adds two additional scripts that helps to parse arguments to pass to SFT Trainer. The `accelerate_launch.py` script is run by default when running the image to trigger SFT trainer for single or multi GPU by parsing arguments and running `accelerate launch launch_training.py`. 
+
+## Configuration
+
+The scripts accept a JSON formatted config which are set by environment variables. `SFT_TRAINER_CONFIG_JSON_PATH` can be set to the mounted path of the JSON config. Alternatively, `SFT_TRAINER_CONFIG_JSON_ENV_VAR` can be set to the encoded JSON config using the below function:
+
+```py
+import base64
+
+ def encode_json(my_json_string):
+    base64_bytes = base64.b64encode(my_json_string.encode("ascii"))
+    txt = base64_bytes.decode("ascii")
+    return txt
+```
+
+The keys for the JSON config are all of the flags available to use with [SFT Trainer](https://huggingface.co/docs/trl/sft_trainer#trl.SFTTrainer).
+
+For configuring `accelerate launch`, use key `multiGPU` and pass the set of flags accepted by [accelerate launch](https://huggingface.co/docs/accelerate/package_reference/cli#accelerate-launch).
+
+For example, the below config is used for running with two GPUs and FSDP for PEFT tuning:
+
+Note that `num_processes` which is the total number of processes to be launched in parallel,should match the number of GPUs to run on.
+
+```json
+{
+    "multiGPU": {
+        "num_machines": 1,
+        "main_process_port": 1234,
+        "num_processes": 2,
+        "use_fsdp": true,
```

Comment:
no value needed here, the flag is passed is as just `--use_fsdp`. Passing it in as `--use_fsdp True` parses incorrectly. hmm how to parse this as a flag appropriately... --> good catch by @tharapalanivel !

Reply:
there are many params like this (search for `action="store_true"` in https://github.com/huggingface/accelerate/blob/4d415f2129f97da4b16f29b9bf0535fe5f2fb0e8/src/accelerate/commands/launch.py) where the flag is passed with no value. How to best handle this for JSON passing in params...

options:
1. Since these params are not exposed to users widely, only allow a subset of the parameters to be used in BAM/SW (num_processes, master_port, etc). Note that if a user deviates from this, the script will break. A workaround is the user can pass all of these params in as their own config_file. This would overwrite our default config_file. 
2. Hard code the set of params that are of this type and parse them accordingly -- only pass in flag and not value.

Thara and I are leaning towards option 1 especially to start but @Ssukriti @fabianlim let us know your thoughts

Reply:
Adding another approach to solve this.

Given that we are getting the cli params in a well structured format (JSON) we can add one more step of cleaning on top of `launch_command_parser` to handle those specific flags that are `_StoreTrueAction` type class. 

You can do specific handling of such arguments/flags from the `accelerate launch` command parser as

```python
from accelerate.commands.launch import launch_command_parser
parser = launch_command_parser()
for action in parser._actions:
    if (type(action).__name__ == "_StoreTrueAction":
    # This way you know which flags are store true actions and then you can simply include 
    # or ignore them in the args list based on the boolean value in the json and they being store true action type or not.
```

@anhuong

Reply:
added this in https://github.com/foundation-model-stack/fms-hf-tuning/pull/98/files#diff-3918b6c0fe504e55b66f13d0cbc2958364e228b0bd7244c67efc5732c5f067f5R58-R63